### PR TITLE
Implement getenv function from os module

### PR DESCRIPTION
**Pull request by dvtkrlbs**
https://docs.python.org/3/library/os.html#os.getenv

### CODE REVIEW

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -1030,6 +1030,26 @@ pub fn os_ttyname(fd: PyIntRef, vm: &VirtualMachine) -> PyResult {
...
+pub struct GetEnvArgs {
+    #[pyarg(positional_or_keyword, optional = false)]
+    key: PyStringRef,
+    #[pyarg(keyword_only, default = "None")]
+    default: Option<PyObjectRef>,
+}
+
+pub fn os_getenv(args: GetEnvArgs, vm: &VirtualMachine) -> PyObjectRef {
+    use std::env;
+    match env::var(&args.key.value) {
```
Comment (isidentical) [inline vm/src/stdlib/os.rs]: i think you should use `os.environ` instead

**Comment by dvtkrlbs** [comment]
Do we need to get these values from os.environ

**Comment by isidentical** [comment]
> Do we need to get these values from os.environ

I believe we should (for cpython compability)
```py
>>> import os
>>> assert os.getenv("A") is None
>>> os.environ["A"] = "B"
>>> assert os.getenv("A") == "B"
>>> os.environ = type("Fake", (), {"get": lambda self, key, default: "blabla"})()
>>> os.getenv("y")
'blabla'
>>> os.getenv("yz")
'blabla'
>>> os.getenv("yzz")
'blabla'
```

**Comment by dvtkrlbs** [comment]
Should be the final version getenv should use os.environ so i fetch os module from sys.modules and then call environ and get the appropriate key. (cpython writes this module in python so their code is more clean)

**Comment by youknowone** [comment]
We have the function in same place: https://github.com/RustPython/RustPython/blob/master/Lib/os.py#L131

Do we need it in the rust module? Reimplementing python stdlibs in rust module sometimes have advantages. But in this case, due to the dynamic fetchings, it seems it doesn't have much advantage.

**Comment by isidentical** [comment]
> Do we need it in the rust module?
@youknowone i dont think we do. CPython also implements `getenv` in python level

**Comment by dvtkrlbs** [comment]
Yes that is right i am not very familiar with the codebase so i completely missed that

**Pull request closed without merging**